src/VisualStudio/Core/Impl/SolutionExplorer/SourceGeneratedFileItems/SourceGeneratedFileItem.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Navigation;
+using Microsoft.Internal.VisualStudio.PlatformUI;
+using Microsoft.VisualStudio.Imaging;
+using Microsoft.VisualStudio.Imaging.Interop;
+
+namespace Microsoft.VisualStudio.LanguageServices.Implementation.SolutionExplorer
+{
+    internal sealed partial class SourceGeneratedFileItem : BaseItem
+    {
+        private readonly string _languageName;
+
+        public SourceGeneratedFileItem(DocumentId documentId, string hintName, string languageName, Workspace workspace)
+            : base(name: hintName)
+        {
+            DocumentId = documentId;
+            HintName = hintName;
+            _languageName = languageName;
+            Workspace = workspace;
+        }
+
+        public DocumentId DocumentId { get; }
+        public string HintName { get; }
+        public Workspace Workspace { get; }
+
+        public override ImageMoniker IconMoniker =>
+            _languageName switch
+            {
+                LanguageNames.CSharp => KnownMonikers.CSFileNode,
+                LanguageNames.VisualBasic => KnownMonikers.VBFileNode,
+                _ => KnownMonikers.Document
+            };
+
+        public override object GetBrowseObject()
+        {
+            return new BrowseObject(this);
+        }
+
+        public override IInvocationController InvocationController => InvocationControllerImpl.Instance;
+
+        private sealed class InvocationControllerImpl : IInvocationController
+        {
+            public static IInvocationController Instance = new InvocationControllerImpl();
+
+            public bool Invoke(IEnumerable<object> items, InputSource inputSource, bool preview)
+            {
+                var item = items.OfType<SourceGeneratedFileItem>().First();
```

Comment:
why the first item?  Should this be single?

Reply:
Ah, um, yeah. I figured you can't have more than one but I'm also wondering what happens if you select two and hit enter...

Reply:
Yep, that was broken. Fixed.